core/cloudflow-flink/src/main/scala/cloudflow/flink/FlinkStreamlet.scala

```diff
@@ -80,24 +80,64 @@ abstract class FlinkStreamlet extends Streamlet[FlinkStreamletContext] with Seri
       streamletDefinition ← StreamletDefinition.read(config)
     } yield {
       val updatedConfig = streamletDefinition.config.withFallback(config)
-      new FlinkStreamletContextImpl(streamletDefinition, setupExecutionEnvironment(createExecutionEnvironment), updatedConfig)
+      new FlinkStreamletContextImpl(streamletDefinition,
+                                    createExecutionEnvironment(updatedConfig, streamletDefinition.streamletRef),
+                                    updatedConfig)
     }).recoverWith {
       case th ⇒ Failure(new Exception(s"Failed to create context from $config", th))
     }.get
 
   /**
-   * Override this method to setup the StreamExecutionEnvironment. By default it does not modify the [[StreamExecutionEnvironment]] created
-   * by [[createExecutionEnvironment]].
-   * @see [[createExecutionEnvironment]] which creates the StreamExecutionEnvironment used in this FlinkStreamlet.
+   * Populate Flink configuration from Streamlet Flink configuration
    */
-  protected def setupExecutionEnvironment(env: StreamExecutionEnvironment): StreamExecutionEnvironment = env
+  private def populateFlinkConfiguration(configuration: Configuration, config: Config, path: String): Boolean = {
```

Comment:
I would love to see this as a method that returns the updated configuration instead of a `Boolean`. The name `populateFlinkConfiguration` always makes me think it will give me an updated configuration.

Reply:
Well, possibly, but...
Boolean here is whether we need WebUI or not. Returning back configuration will make code a bit more verbose and will use more memory

Reply:
I think @debasishg  has a point here, it would be nicer to return a new version of the config, you can also return any control information and use it like `val (newConfig, needWebUI) = populateFlinkConfiguration(...)`